pulp_ansible/app/tasks/collections.py

```diff
@@ -608,13 +663,55 @@ async def _find_all_collections(self):
 
         await asyncio.gather(*tasks)
 
+    async def _can_be_optimized(self):
+        """Check last synced metadata time."""
+        root, api_version = await self._get_root_api(self.remote.url)
+        if api_version == 3:
+            downloader = self.remote.get_downloader(
+                url=root, silence_errors_for_response_status_codes={404}
+            )
+            try:
+                metadata = parse_metadata(await downloader.run())
+            except FileNotFoundError:
+                return False
+
+            self.last_synced_metadata_time = parse_datetime(metadata.get("published", ""))
```

Comment:
In cases where there is no published I think the op-op must return False. So we can be more explicit here with:
```
try:
    self.last_synced_metadata_time = parse_datetime(metadata["published"])
except KeyError:
    return False
```